File: src/main/java/su/nsk/iae/post/dsm/manager/common/ServerUtils.java
package su.nsk.iae.post.dsm.manager.common;

import java.io.IOException;
import java.net.ServerSocket;

public class ServerUtils {

    public static int findFreePort() {
        try (ServerSocket socket = new ServerSocket(0)) {
            socket.setReuseAddress(true);
            return socket.getLocalPort();
        } catch (IOException e) {
            Logger.error(ServerUtils.class, e.getMessage());
        }
        throw new IllegalStateException("Could not find a free TCP/IP port");
    }
}

File: post-ide-dsm-manager/src/main/kotlin/su/nsk/iae/post/dsm/manager/common/ServerUtils.kt
package su.nsk.iae.post.dsm.manager.common

import su.nsk.iae.post.dsm.manager.common.Logger.error
import java.io.IOException
import java.net.ServerSocket

object ServerUtils {

    fun findFreePort(): Int {
        try {
            ServerSocket(0).use { socket ->
                socket.reuseAddress = true
                return socket.localPort
            }
        } catch (e: IOException) {
            error(ServerUtils::class.java, e.message!!)
        }
        throw IllegalStateException("Could not find a free TCP/IP port")
    }
}